components/newsletter-sign-up/newsletter-sign-up.jsx

```diff
@@ -0,0 +1,288 @@
+import React from "react";
```

Comment:
Pretty much the same as https://github.com/mozilla/foundation.mozilla.org/blob/master/source/js/components/join/join.jsx with some minor changes (doesn't render name fields as Pulse doesn't need it)